integration/hibernate-4.2/src/main/java/com/blazebit/persistence/integration/hibernate/Hibernate4Access.java

```diff
@@ -314,4 +331,40 @@ public QueryParameters createQueryParameters(
     public ParameterTranslations createParameterTranslations(List<ParameterSpecification> queryParameterSpecifications) {
         return new ParameterTranslationsImpl(queryParameterSpecifications);
     }
+
+    /**
+     * Iterator over {@code ScrollableResults}.
+     *
+     * @author Jan-Willem Gmelig Meyling
+     * @since 1.6.2
+     */
+    private static class ScrollableResultsIterator implements Iterator<Object> {
+        private final ScrollableResults scrollableResults;
+
+        public ScrollableResultsIterator(ScrollableResults scrollableResults) {
+            this.scrollableResults = scrollableResults;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return scrollableResults.next();
+        }
+
+        @Override
+        public Object next() {
+            Object[] next = scrollableResults.get();
+            if (next.length == 1) {
+                return next[0];
+            } else {
+                return next;
+            }
+        }
+
+        @Override
+        public void remove() {
```

Comment:
Remove?

Reply:
Remove was still mandatory in 1.7, and Hibernate 5 doesn't target 1.8.